### PR TITLE
Simple fallback strategy

### PR DESCRIPTION
**Pull request by jmccann**
Let me know your thoughts @donny-dont

### CODE REVIEW

File: cache/cache.go [hunk 1]
```diff
@@ -54,7 +48,7 @@ func (c Cache) Restore(src string) error {
 	return nil
 }
 
-func restoreCache(src string, s storage.Storage, a archive.Archive) error {
+func restoreCache(src string, fallback string, s storage.Storage, a archive.Archive) error {
```
Comment (donny-dont) [inline cache/cache.go]: I feel like `restoreCache` should be dumb and just try and grab the file and restore it and the fallback should happen in `Restore`.

**Review by donny-dont — CHANGES_REQUESTED** [review]

File: cache/cache.go (hunk 1 above)
Comment (jmccann) [reply cache/cache.go]: Sounds good to me!  I'll get on it!

File: cache/cache.go [hunk 2]
```diff
@@ -63,7 +57,14 @@ func restoreCache(src string, s storage.Storage, a archive.Archive) error {
 	go func() {
 		defer writer.Close()
 
-		cw <- s.Get(src, writer)
+		err := s.Get(src, writer)
+
+		if err != nil && fallback != "" {
```
Comment (donny-dont) [inline cache/cache.go]: Also feel like you should check to see if the fallback points to the same file before trying to run the fallback logic.

File: cache/cache.go (hunk 1 above)
Comment (donny-dont) [reply cache/cache.go]: I think that'll make things easier if we do end up having a more robust fallback strategy. Like if you wanted it to search for the branch this branch was forked from.

**Comment by jmccann** [comment]
@donny-dont I think I got you concerns covered.

**Review by donny-dont — APPROVED** [review]

**Pull request merged**